libwallet/assets_manager.go

```diff
@@ -979,7 +981,10 @@ func (mgr *AssetsManager) DeleteDEXData() error {
 	// both will get the ntfn?
 	<-shutdownChan // wait for shutdown
 
-	// TODO: Set mgr.dexc to nil and unregister the custom wallet constructors!
+	mgr.dexcMtx.Lock()
+	mgr.dexc = nil
+	// TODO: Also unregister the custom wallet constructors!
```

Comment:
Let's take off this TODO now. The wallet constructors are now safely initialized just once for the lifetime of the app and no longer need to be unregistered.